modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManager.java

```diff
@@ -343,6 +390,61 @@ public SCIMResponse listWithGET(UserManager userManager, String filter, int star
         }
     }
 
+    /**
+     * Method to process a list and return a SCIM response.
+     *
+     * @param tempList          Filtered user list
+     * @param encoder           Json encoder
+     * @param attributes        Required attributes
+     * @param excludeAttributes Exclude attributes
+     * @param startIndex        Starting index
+     * @return SCIM response
+     * @throws NotFoundException
+     * @throws CharonException
+     * @throws BadRequestException
+     */
+    private SCIMResponse processGroupList(List<Object> tempList, JSONEncoder encoder, String attributes,
+            String excludeAttributes, int startIndex) throws NotFoundException, CharonException, BadRequestException {
+
+        int totalResults = 0;
+        List<Object> returnedGroups;
+        if (tempList == null) {
+            tempList = Collections.emptyList();
+        }
+        try {
+            totalResults = (int) tempList.get(0);
+            tempList.remove(0);
+        } catch (IndexOutOfBoundsException e) {
+            if (logger.isDebugEnabled()) {
```

Comment:
shall we rewrite the logic? inline 411 we do null check and assign empty then again here having IndexOutOfBoundsException, we can do the empty check as CollectionUtils.isEmpty at the beginning?

Reply:
apache-commons in not in Charon. Therefore, unable to user StringUtils.